srcpkgs/texlive/files/tlmgr

```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+$(dirname $(realpath -s $0))/../share/texmf-dist/scripts/texlive/tlmgr.pl $@
```

Comment:
Why not just `/usr/share` ? And `$@` should be quoted, so `"$@"`.

I dislike this as a solution, tho. The texlive-bin packages have it working, so what are we doing differently?

Reply:
There is no gurantee it will be `/usr`.

> The texlive-bin packages have it working, so what are we doing differently?

Literally *everything*. Compiling TeXLive from source is a completely different beast to the distributed installations. TBH, this is essentially the wrong question.

I don't disagree that it is a bad solution, though... still looking for a btter solution that uses kpsewhich like the rest of the TeX packages do.

Reply:
The problem lies here in `texmf.cnf` and `tlmgr`. First, in `texmf.cnf`:

```
% ^L Part 1: Search paths and directories.

% This is the parent directory of our several trees, i.e.,
% /usr/local/texlive/YYYY in the original TeX Live distribution.
%
% The path definitions here assume the directories are organized
% according to the TeX Directory Structure (http://tug.org/tds).
%
% Redistributors will probably want $SELFAUTODIR/share, i.e., /usr/share.
% Kpathsea sets (in kpathsea/progname.c):
% SELFAUTOLOC (directory of the binary=/usr/local/texlive/YYYY/bin/i386-linux),
% SELFAUTODIR (its parent = /usr/local/texlive/YYYY/bin),
% SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
% SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
```

`tlmgr` has:
```
  if (-r "$bindir/$kpsewhichname") {
    # if not in bootstrapping mode => kpsewhich exists, so use it to get $Master
    chomp($Master = `kpsewhich -var-value=SELFAUTOPARENT`);
  }
```

Hence because `SELFAUTLOC=/usr/bin` `SELFAUTOPARENT=/` and so it adds `/tlpkg` and `/texmf-dist/scripts/texlive` to `@INC` and can't find anything. These variable definitions are hardcoded into kpathsea so they cannot be changed; `SELFAUTOPARENT` will always be `/`. So, the three options are a wrapper like this one, patching tlmgr to use `/usr/share/texmf-dist` as `$Master` or patching kpathsea. None of these are all that nice, IMHO.

Reply:
> There is no gurantee it will be /usr.

Why not? We are the ones putting the script there.

> % Redistributors will probably want $SELFAUTODIR/share, i.e., /usr/share.

Doesn't this mean that this is a known usecase, and that they are ready for it? It feels like it should require config options instead of patching :/ 

Overall, a wrapper script is better than patching, if only because it avoids rebasing for a new release.

Reply:
TeX only officially supports https://tug.org/tds/tds.html the TDS. Its kinda dumb IMO; they expect **all** things, including its scripts such as tlmgr, to be in its own self-contained tree which doesn't adhere to FHS. Hence they by default expect binaries to be in this tree. All Linux distributions that I know of ignore the TDS for binaries and adhere to `/usr/bin` for binaries as per the FHS. This is why `texlive-bin` works, it has the TDS in `/opt/texlive/YYYY`.

> Doesn't this mean that this is a known usecase, and that they are ready for it?

The line you quotes pertains to the line after that I missed in my code excerpt, about `TEXMFROOT` (which it expects its own scripts to be).

> Why not? We are the ones putting the script there.

Mm, I guess. I'll change it.

Reply:
> Its kinda dumb IMO; they expect all things, including its scripts such as tlmgr, to be in its own self-contained tree which doesn't adhere to FHS

lol, sounds great. Ok then!